whitepapers/tickets/SHARE-822.md

```diff
@@ -0,0 +1,112 @@
+# [SHARE-822] Custom Taxonomies
+
+
+## Goals:
+   - Support OSF's Custom Taxonomies requirements
+   - Have a solution that can be used by any parties interested in enhancing SHARE
+   - Migrate the existing Subjecting system to this new solution
+   - Continue to use the Bepress taxonomy as SHARE's central taxonomy
+
+
+## Changes to SHARE:
+  - New Model: "Taxonomy"
+  - Changes to Model: "Subject"
+    - Unique indexes on Subjects will move from `[name]` to `[name, taxonomy_id]` if the OSF requirements are ammenable to this
+      Otherwise it will become `[name, taxonomy_id, parent_id]`
+  - If we can upgrade to elasticsearch 5.4+ in time SHARE will index subjects using the new Path Hierarchy Tokenizer.
```

Comment:
[2.x has that, too](https://www.elastic.co/guide/en/elasticsearch/reference/2.3/analysis-pathhierarchy-tokenizer.html). We should still upgrade to 5.x, though.